.github/workflows/publish_branch.yaml

```diff
@@ -0,0 +1,101 @@
+name: Publish all ROCKs from selected branch
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'Which branch to publish'
+        required: true
+        default: main
+        type: choice
+        options:
+        - main
+        - stable/2023.1
+
+
+jobs:
+  allrocks:
+    runs-on: ubuntu-latest
+    outputs:
+      rocks: ${{ steps.all-rocks.outputs.rocks }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.branch }}
+      - name: List rockcraft files
+        id: list-files
+        run: |
+          all_rock_files=$(find rocks -name rockcraft.yaml -printf "%p ")
+          echo "All rock files: $all_rock_files"
+          echo "all_rock_files=$all_rock_files" >> $GITHUB_OUTPUT
+      - name: set output
+        id: all-rocks
+        run: |
+          components=()
+          # trim file name to component name
+          for file in ${{ steps.list-files.outputs.all_rock_files }}; do
+              component=$(echo $file | sed 's/rocks\/\(\S.*\)\/rockcraft.yaml/\1/')
+              [[ ! -z "$component" ]] && components+=($component)
+          done
+          all_rocks=$(jq --compact-output --null-input '$ARGS.positional' --args -- "${components[@]}")
+          echo "All rocks: $all_rocks"
+          echo "rocks=$all_rocks" >> $GITHUB_OUTPUT
+  build:
+    needs: allrocks
+    if: ${{ needs.allrocks.outputs.rocks != '[]' }}
+    strategy:
+      matrix:
+        rock: ${{ fromJson(needs.allrocks.outputs.rocks) }}
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.branch }}
+      - uses: canonical/craft-actions/rockcraft-pack@main
+        id: rockcraft
+        with:
+          path: rocks/${{ matrix.rock }}
+      - uses: actions/upload-artifact@v3
```

Comment:
I am just thinking whether we can avoid upload here and download again in publish step.

Reply:
Yeah, I'll make publish a step, and comply to the matrix